ql/cashflows/cpicoupon.cpp

```diff
@@ -186,6 +263,22 @@ namespace QuantLib {
         }
     }
 
+    Real CPICashFlow::amount() const {
+        Rate I0 = baseFixing();
+
+        // If BaseFixing is null, use the observed index fixing
+        if (I0 == Null<Rate>()) {
+            I0 = IndexedCashFlow::baseFixing();
+        }
```

Comment:
I would move this logic inside the `baseFixing` method.  The same goes for the coupon, which would simplify `CPICouponPricer::adjustedFixing` as well.

Reply:
I understand this point, but I would like to have the possibility to inspect the user specified baseFixing / baseCPI value as well. If we change the logic we can not see if thats a index fixing or a user specified value.

If I call baseFixing() before loading the fixing, I will see an error even I dont want to price the instrument yet. 

Can we made an extra method in the class or utility function to derive the baseFixing during pricing?

Reply:
Or add a Boolean parameter useBaseDateAsFallback to the baseFixing method (defaulted to false) and if it’s false the method returns the baseFixing and if it’s true and baseFixing is Null it uses the fixing from the index?

Reply:
I see.  Oh well, the logic is in just one place anyway, let's keep it as it is now.